src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1494,34 +1495,46 @@
   "Delete all resource events in the database by dropping any partition older than the day of the year of the given
   date.
   Note: this ignores the time in the given timestamp, rounding to the day."
-  [date incremental?]
+  [date incremental? update-lock-status]
   {:pre [(kitchensink/datetime? date)]}
-  (prune-daily-partitions "resource_events" date incremental?))
+  (update-lock-status :write-locking-resource-events inc)
+  (try!
```

Comment:
In case prune throws an exception, and then update-lock-status does too.  We'll at least have a record of what actually happened in the backtrace, i.e. we won't lose the first exception.

Reply:
:+1: